src/metabase/util/stats.clj

```diff
@@ -0,0 +1,312 @@
+
+;; util function
+(def add-summaries
+  "add up some dictionaries"
+  (partial merge-with +))
+
+;; User metrics
+(defn- user-dims
+  "characterize a user record"
+  [user]
+  {:total 1
+   :active (if (user :is_active) 1 0) ;; HOW DO I GET THE LIST OF ALL USERS INCLUDING INACTIVES?
+   :admin (if (user :is_superuser) 1 0)
+   :logged-in (if (nil? (user :last_login)) 0 1)
+   :sso (if (nil? (user :google_auth)) 0 1)})
+
+
+(defn- user-metrics
+  "Get metrics based on user records
+  TODO: get activity in terms of created questions, pulses and dashboards"
+  []
+  (let [users (db/select 'User)]
+    {:users (apply add-summaries (map user-dims users))}))
+
+
+(defn- group-metrics
+  "Get metrics based on groups:
+  TODO characterize by # w/ sql access, # of users, no self-serve data access"
+  []
+  (let [groups (db/select 'PermissionsGroup)]
+    {:groups (count groups)}))
+
+;; Artifact Metrics
+(defn- question-dims
+  "characterize a saved question
+  TODO: characterize by whether it has params, # of revisions, created by an admin"
+  [question]
+    {:total 1
+     :native (if (= (question :iquery_type) "native") 1 0)
+     :gui (if (not= (question :iquery_type) "native") 1 0)})
+
+(defn- question-metrics
+  "Get metrics based on questions
+  TODO characterize by # executions and avg latency"
+  []
+  (let [questions (db/select 'Card)]
+    {:questions (apply add-summaries (map question-dims questions))}))
+
+(defn- dashboard-metrics
+  "Get metrics based on dashboards
+  TODO characterize by # of revisions, and created by an admin"
+  []
+  (let [dashboards (db/select 'Dashboard)
+        dashcards (db/select 'DashboardCard)]
+    {:dashboards (count dashboards)
+     :num_dashs_per_user (medium-histogram dashboards :creator_id)
+     :num_cards_per_dash (medium-histogram dashcards :dashboard_id)
+     :num_dashs_per_card (medium-histogram dashcards :card_id)}))
+
+(defn- pulse-metrics
+  "Get mes based on pulses
+  TODO: characterize by non-user account emails, # emails"
+  []
+  (let [pulses (db/select 'Pulse)
+        pulsecards (db/select 'PulseCard)
+        pulsechannels (db/select 'PulseChannel)]
+    {:pulses (count pulses)
+     :pulse-types (frequencies (map :channel_type pulsechannels))
+     :pulse-schedules (frequencies (map :schedule_type pulsechannels))
+     :num_pulses_per_user (medium-histogram pulses :creator_id)
+     :num_pulses_per_card (medium-histogram pulsecards :card_id)
+     :num_cards_per_pulses (medium-histogram pulsecards :pulse_id)}))
+
+
+(defn- label-metrics
+  "Get metrics based on labels"
+  []
+  (let [labels (db/select 'CardLabel)]
+    {:labels (count labels)
+     :num_labels_per_card (micro-histogram labels :card_id)
+     :num_cards_per_label (medium-histogram labels :label_id)}))
+
+;; Metadata Metrics
+(defn- database-dims
+  "characterize a database record"
+  [database]
+  {:total 1
+   :analysed (if (:is_full_sync database) 1 0)})
```

Comment:
Any reason we're using British spelling here instead of `:analyzed`? I'm worried someone will accidentally assume American spelling somewhere else (as we do in the rest of the codebase) and we'll end up with duplicate keys.

If that's what the server expects we should add a note